### PR TITLE
net/lwip: support window scale option

### PR DESCRIPTION
**Pull request by pillip8282**
support window scale option

### CODE REVIEW

File: os/net/lwip/configs/tcp/Kconfig [hunk 1]
```diff
@@ -18,6 +18,23 @@ config NET_TCP_WND
...
+	---help---
+		Set LWIP_WND_SCALE to 1 to enable window scaling.
+
+
+if NET_WND_SCALE
+config NET_TCP_RCV_SCALE
+	int "Receive window scale support"
+	default 0
+	---help---
+		Set TCP_RCV_SCALE to the desired scaling factor (shift count in the range of [0..14]).
```
Comment (jeongchanKim) [inline os/net/lwip/configs/tcp/Kconfig:33]: If this config has value range, how about adding `range a b`?
Then we can prevent invalid value.

File: os/net/lwip/src/include/lwip/lwipopts.h [hunk 2]
```diff
@@ -440,6 +440,14 @@
 #define TCP_WND_UPDATE_THRESHOLD	CONFIG_NET_TCP_WND_UPDATE_THRESHOLD
 #endif
 
+#ifdef CONFIG_NET_WND_SCALE
+#define LWIP_WND_SCALE 1
+#endif
+
+#ifdef CONFIG_NET_TCP_RCV_SCALE
+#define TCP_RCV_SCALE CONFIG_NET_TCP_RCV_SCALE
```
Comment (davidfather) [inline os/net/lwip/src/include/lwip/lwipopts.h:448]: If CONFIG_NET_TCP_RCV_SCALE is not set, what is TCP_RCV_SCALE?

File: os/net/lwip/configs/tcp/Kconfig (hunk 1 above)
Comment (pillip8282) [reply os/net/lwip/configs/tcp/Kconfig:33]: Thanks. It's better to reduce mistake.

File: os/net/lwip/src/include/lwip/lwipopts.h (hunk 2 above)
Comment (pillip8282) [reply os/net/lwip/src/include/lwip/lwipopts.h:448]: it's defined to opt.h. default value is 0

**Review by an4967 — APPROVED** [review]
LGTM

**Pull request merged**